Core/OrderHelper.php

```diff
@@ -197,6 +200,33 @@ public static function updateOrderState($oOrder, $oResponse, $oBackendService)
         $oOrder->save();
     }
 
+    /**
+     * Manages Pia Payment Information, for the later use on Thank You page
+     *
+     * @param Response $oResponse
+     * @param Order    $oOrder
+     *
+     * @since 1.3.0
+     */
+    private static function _managePiaPaymentInformation($oResponse, $oOrder)
+    {
+        if ($oOrder->oxorder__oxpaymenttype->value === "wdpaymentinadvance") {
+            $oResponseXml = simplexml_load_string($oResponse->getRawData());
+
+            $oSession = Registry::getSession();
+            $oSession->setVariable(
+                "wdPaymentInAdvancePaymentInformation",
+                new PaymentInAdvancePaymentInformation(
+                    (string) $oResponseXml->{'requested-amount'} . ' ' .
```

Comment:
The first cast to `string` is not necessary because the concatenation converts it to a string anyway.

Reply:
Nice 👍